### PR TITLE
OCM-1733 | feat: list instance-types by region

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

**Comment by thomasmckay** [comment]
@gdbranco How do you wish to proceed on this? I cannot retrieve the regions without WithAWS() to validate that it exists (happy to be corrected here). If the region doesn't exist (or cannot be logged into) the command will hang.

File: cmd/list/instancetypes/cmd.go [hunk 1]
```diff
@@ -24,57 +24,114 @@ import (
...
-	Run: run,
+var args struct {
+	searchRegion string
+}
+
+const (
+	SearchRegionFlag = "limit-to-region"
+)
+
+var Cmd = makeCmd()
```
[the remark this answers is not in the document]
Comment (thomasmckay) [reply cmd/list/instancetypes/cmd.go:41]: this is the pattern in use for allowing testing. we can refactor it everywhere, if desired, in a separate jira.

File: cmd/list/instancetypes/cmd.go [hunk 2]
```diff
@@ -24,57 +24,114 @@ import (
...
+
+	return cmd
 }
 
 func init() {
-	output.AddFlag(Cmd)
+	initFlags(Cmd)
+}
+
+func initFlags(cmd *cobra.Command) {
```
[the remark this answers is not in the document]
Comment (thomasmckay) [reply cmd/list/instancetypes/cmd.go:61]: this is the pattern in use for allowing testing. we can refactor it everywhere, if desired, in a separate jira.

File: pkg/aws/client.go [hunk 3]
```diff
@@ -65,6 +65,8 @@ import (
 )
 
 // Name of the AWS user that will be used to create all the resources of the cluster:
+//
+//go:generate mockgen -source=client.go -package=aws -destination=mock_client.go
```
Comment (thomasmckay) [inline pkg/aws/client.go:69]: @robpblake if i set package=mocks and destination=mocks/client.go i get errors about cycle imports in all the other pkg/aws files

File: cmd/list/instancetypes/cmd_test.go [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+			IsSTS:     false,
+		}
+		DeferCleanup(r.Cleanup)
+	})
+
+	AfterEach(func() {
+		ssoServer.Close()
+		apiServer.Close()
+	})
+
```
[the remark this answers is not in the document]
Comment (thomasmckay) [reply cmd/list/instancetypes/cmd_test.go:242]: a couple more tests added

**Review by robpblake — APPROVED** [review]
/lgtm

**Pull request merged**